src/hooks/use-deep-compare/use-deep-compare.tsx

```diff
@@ -0,0 +1,29 @@
+import * as React from 'react';
+import isEqual from 'lodash/isEqual';
+
+type EffectCallback = () => void | (() => void | undefined);
+type DependencyList = ReadonlyArray<unknown>;
+type Comparator = (a: DependencyList, b: DependencyList) => boolean;
+
+function useDeepCompareRef(
+  dependencies: DependencyList,
+  comparator: Comparator = isEqual,
```

Comment:
We won't always want to use isEqual, for example in options list we want to compare arrays and our method is cheaper.